tests/test_env.py

```diff
@@ -79,13 +79,24 @@ def test_basic_install(tmpdir):
     import iplib  # NOQA
 
 
-def test_install_fails(tmpdir, caplog):
+def test_install_fails(tmpdir, caplog, monkeypatch):
     venv = env.VirtualEnv(tmpdir)
     venv.use_virtual_env()
     caplog.clear()
     caplog.set_level(logging.INFO)
     package_name = "non-existing-pkg-inmanta"
 
+    # monkeypatch pip install to set --no-index for security reasons (anyone could publish this package to PyPi)
+    compose = env.PipCommandBuilder.compose_install_command
+
+    def mock_compose(*args, **kwargs):
+        if "index_urls" in kwargs or len(args) < 5:
+            return compose(*args, **{**kwargs, "index_urls": []})
+        else:
+            return compose(*args[:5], [], *args[5:], **kwargs)
```

Comment:
After testing it turns out you're right but I must confess I don't get it. I made two mistakes
1. I forgot to exclude the existing index url argument, this part I get
2. I got the count wrong, this I don't get. Why does `cls` not show up in args? Because `compose_mock` is not a classmethod?

Anyway, it will definitely do, I'm just a bit confused.